test/jdk/java/nio/file/DirectoryStream/SecureDS.java

```diff
@@ -172,6 +177,62 @@ static void doBasicTests(Path dir) throws IOException {
         delete(dir2);
     }
 
+    // Exercise setting permisions on the SecureDirectoryStream's view
+    static void doSetPermissions(Path dir) throws IOException {
+        Path aDir = createDirectory(dir.resolve("dir"));
+
+        Set<PosixFilePermission> noperms = EnumSet.noneOf(PosixFilePermission.class);
+        Set<PosixFilePermission> permsDir = getPosixFilePermissions(aDir);
+
+        SecureDirectoryStream<Path> stream =
```

Comment:
If you are doing any more edits then you can change this to use try-with-resources.